### PR TITLE
[CBRD-20203] Insert in btree overflow OID page adds new object instead of replacing existing one

### PR DESCRIPTION
**Pull request by razvanradulescu**

### CODE REVIEW

File: src/storage/btree.c [hunk 1]
```diff
@@ -12222,7 +12224,11 @@ btree_find_oid_from_ovfl (THREAD_ENTRY * thread_p, BTID_INT * btid_int, PAGE_PTR
 	      *offset_to_object = CAST_BUFLEN (oid_ptr - ovf_record.data);
 	    }
 	  /* Not in this page. */
```
Comment (eseokoh) [inline src/storage/btree.c]: @razvanradulescu 

```C
          (void) btree_unpack_mvccinfo (ptr, mvcc_info, BTREE_OID_HAS_MVCC_INSID_AND_DELID);
          error_code = btree_find_oid_does_mvcc_info_match (thread_p, mvcc_info, purpose, match_mvccinfo, &is_match);
          if (error_code != NO_ERROR)
            { 
              ASSERT_ERROR ();
              return error_code;
            }
          if (is_match)
            {
              /* Object is a match. */
              *offset_to_object = CAST_BUFLEN (oid_ptr - ovf_record.data);
            }
          /* Not in this page. */
```

I think the above snippets are useless. Please confirm.

Comment (eseokoh) [inline src/storage/btree.c]: @razvanradulescu 

I think I understand your intention. It seems you wanted `btree_seq_find_oid_from_ovfl` to check only the left side and the right side of oid and the above misses `return` for matched case.
```C
  if (is_match)
    {
      ...
      return NO_ERROR;
    }
```

I'd like to propose to remove matching from `btree_find_oid_from_ovfl` and change `btree_seq_find_oid_from_ovfl` to check matching from the `mid` position. So, the change would be as follows:
```C
  /* first, check previous OIDs */
  oid_ptr = initial_oid_ptr - obj_size;
```

->

```C
  /* first, check OID and previous ones */
  oid_ptr = initial_oid_ptr;
```

**Comment by razvanradulescu** [comment]
Change to remove duplicate code.

**Comment by eseokoh** [comment]
@razvanradulescu 

It looks great. 
Would you merge the patch? 😃

**Pull request merged**